packages/preferences/src/browser/util/preference-event-service.ts

```diff
@@ -24,4 +24,5 @@ export class PreferencesEventService {
     onEditorScroll = new Emitter<Preference.MouseScrollDetails>();
     onNavTreeSelection = new Emitter<Preference.SelectedTreeNode>();
     onDisplayChanged = new Emitter<boolean>();
+    onPreferenceReset = new Emitter<string>();
```

Comment:
When we created this `PreferenceEventService` for #7105, Anton wasn't a big fan because of how it separates emitters from the logical source of emission, and I'm trying to eliminate it in #8263. Could the same functionality be achieved by hooking into the `PreferenceService.onPreferenceChanged` logic and listeners?

Reply:
@colin-grant-work Thank you for the review. I agree your point is valid, the reason why I created a new emitter and didn't use the `preferenceService.onPreferenceChanged` was because I am unaware as to how will I be able to distinguish a `setPreference` done by `RESET_SETTINGS` against a `setPreference` done when there's a change in preference.  Do you have any suggestions?

Reply:
I have updated the code to use `onPreferenceChanged` instead of using a new emitter to fix the issue.

Reply:
Excellent, I think this is a much better style. There is a bit of a hitch, though. Since you validate the input before setting the preference (a good idea), setting an invalid value and then clicking 'reset preferences' is a no-op, because the change of preference never happened, so the reset doesn't trigger an event.

(You can test this with the `editor.fontSize` preference. By default it's `14`, and deleting the `4` or both digits produces unacceptably small values, so no preference change is triggered, and clicking reset preference won't do anything, either.)

Given that wrinkle, there isn't really an external stimulus to hook into unless we create one, so maybe using the `PreferenceEventService` is OK for now, and figuring out a better way to handle it can wait for whoever tries to get rid of the `PreferenceEventService` entirely.

Reply:
@Anasshahidd21 @colin-grant-work what do you think of performing a reset (to the last valid value) during the `onBlur` instead of using the `preference-service` at all. It will cover all cases, and I believe it is fine to assume that if an end-user dismisses an `input` which has errors we should revert their changes to the actual value:

<div align='center'>

![pref-validation](https://user-images.githubusercontent.com/40359487/88982130-78fc9280-d295-11ea-8aeb-88b5938f3ac6.gif)


</div>

Reply:
Yes, very good idea!